microservices/lib/redshift.py

```diff
@@ -11,6 +12,29 @@
 class RedShift:
     'Common microservice operations for RedShift'
 
+    def print_psycopg2_exception(self, err):
+        'handles and parses psycopg2 exceptions'
+        # get details about the exception
+        err_type, err_obj, traceback = sys.exc_info()
+
+        # get the line number when exception occured
+        line_num = traceback.tb_lineno
+
+        # print the connect() error
+        self.logger.error("psycopg2 %s: %s", err.__class__.__name__, err)
+        self.logger.debug("pgerror: %s", err.pgerror)
+        self.logger.debug("pgcode: %s", err.pgcode)
+        self.logger.debug("psycopg2 error on line number: %s", line_num)
+        self.logger.debug("psycopg2 traceback: %s", traceback)
+        self.logger.debug("psycopg2 error type: %s", err_type)
+
+        if str(err.pgcode) == 'XX000':
```

Comment:
Can you add a quick comment here about why we handle XX000 specially?

Reply:
done.